### PR TITLE
[extractor/common] Fix crash in _report_ignoring_subs

### PR DESCRIPTION
**Pull request by salty-horse**
### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [ ] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

When subs are not available and `_report_ignoring_subs` is called, it crashes because it passes an argument to `bug_reports_message`, which accepts none.

I changed it to be appended to the message, like what's done in other places.
Now the message is like this:

> WARNING: [youtube] Ignoring subtitle tracks found in the DASH manifest; if any subtitle tracks are missing, ; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

The ", ;" is not correct but I don't know if you want to get rid of the semicolon. The important thing is that it doesn't crash, and the file can be downloaded.

### CODE REVIEW

**Comment by dirkf** [comment]
Thanks, good catch.

There was supposed to be a matching change to `utils.py`: I'd prefer to add that rather than change the code in `extractor/common.py`.

Would you like to revert your change and apply the missing change (adapted from yt-dlp) instead?

```diff
--- old/youtube-dl/youtube_dl/utils.py
+++ new/youtube-dl/youtube_dl/utils.py
@@ -2371,15 +2371,24 @@
         return YoutubeDLHTTPSHandler(params, context=context, **kwargs)
 
 
-def bug_reports_message():
+def bug_reports_message(before=';'):
     if ytdl_is_updateable():
         update_cmd = 'type  youtube-dl -U  to update'
     else:
-        update_cmd = 'see  https://yt-dl.org/update  on how to update'
-    msg = '; please report this issue on https://yt-dl.org/bug .'
-    msg += ' Make sure you are using the latest version; %s.' % update_cmd
-    msg += ' Be sure to call youtube-dl with the --verbose flag and include its complete output.'
-    return msg
+        update_cmd = 'see  https://github.com/ytdl-org/youtube-dl/#user-content-installation  on how to update'
+
+    msg = (
+        ' please report this issue on https://github.com/ytdl-org/youtube-dl/issues ,'
+        ' using the appropriate issue template.'
+        ' Make sure you are using the latest version; %s.'
+        ' Be sure to call youtube-dl with the --verbose option and include the complete output.'
+    ) % update_cmd
+
+    before = (before or '').rstrip()
+    if not before or before.endswith(('.', '!', '?')):
+        msg = msg[0].title() + msg[1:]
+
+    return (before + ' ' if before else '') + msg
 
 
 class YoutubeDLError(Exception):
```

**Comment by salty-horse** [comment]
That makes much more sense!

I'm uncomfortable with copy-pasting someone else's code without credit - I'd much rather you commit it as originally intended.

File: youtube_dl/extractor/common.py [hunk 1]
```diff
@@ -1812,10 +1812,10 @@ def _m3u8_meta_format(self, m3u8_url, ext=None, preference=None, m3u8_id=None):
         }
 
     def _report_ignoring_subs(self, name):
-        self.report_warning(bug_reports_message(
+        self.report_warning(
             'Ignoring subtitle tracks found in the {0} manifest; '
-            'if any subtitle tracks are missing,'.format(name)
-        ), only_once=True)
+            'if any subtitle tracks are missing, {1}'.format(name, bug_reports_message()),
+            only_once=True)
```
Comment (dirkf) [inline youtube_dl/extractor/common.py:1818]: Revert and fix `utils.bug_reports_message()` (1):
```suggestion
        self.report_warning(bug_reports_message(
            'Ignoring subtitle tracks found in the {0} manifest; '
            'if any subtitle tracks are missing,'.format(name)
        ), only_once=True)
```

File: youtube_dl/utils.py [hunk 2]
```diff
@@ -2371,15 +2371,12 @@ def set_alpn_protocols(ctx):
...
+def bug_reports_message(before=';'):
     if ytdl_is_updateable():
         update_cmd = 'type  youtube-dl -U  to update'
     else:
         update_cmd = 'see  https://yt-dl.org/update  on how to update'
-    msg = '; please report this issue on https://yt-dl.org/bug .'
-    msg += ' Make sure you are using the latest version; %s.' % update_cmd
-    msg += ' Be sure to call youtube-dl with the --verbose flag and include its complete output.'
-    return msg
+    return '; please report this issue on https://yt-dl.org/bug .'
```
Comment (dirkf) [inline youtube_dl/utils.py:2379]: Revert and fix `utils.bug_reports_message()` (2):
```suggestion
    else:
        update_cmd = 'see  https://github.com/ytdl-org/youtube-dl/#user-content-installation  on how to update'

    msg = (
        ' please report this issue on https://github.com/ytdl-org/youtube-dl/issues ,'
        ' using the appropriate issue template.'
        ' Make sure you are using the latest version; %s.'
        ' Be sure to call youtube-dl with the --verbose option and include the complete output.'
    ) % update_cmd

    before = (before or '').rstrip()
    if not before or before.endswith(('.', '!', '?')):
        msg = msg[0].title() + msg[1:]

    return (before + ' ' if before else '') + msg
```

**Review by dirkf — COMMENTED** [review]
With my maintainer superpower, I (will) have added some commits into your PR branch to make my suggested change.

It would be a great help if you'd be able to pull the latest version of the branch and test that the issue is fixed (since I'm deep in the mind-numbing twaddle that fixing the TikTok extractor brings). 

Once it's OK, I can merge the PR as a single commit credited to us jointly with acknowledgments to yt-dlp, and then it can be in the next nightly build.

File: youtube_dl/extractor/common.py (hunk 1 above)
Comment (dirkf) [inline youtube_dl/extractor/common.py:1818]: Revert and fix `utils.bug_reports_message()` (1b):
```suggestion
            'if any subtitle tracks are missing,'.format(name)
        ), only_once=True)
```

File: youtube_dl/utils.py [hunk 3]
```diff
@@ -2371,15 +2371,12 @@ def set_alpn_protocols(ctx):
         return YoutubeDLHTTPSHandler(params, context=context, **kwargs)
 
 
-def bug_reports_message():
+def bug_reports_message(before=';'):
     if ytdl_is_updateable():
         update_cmd = 'type  youtube-dl -U  to update'
     else:
         update_cmd = 'see  https://yt-dl.org/update  on how to update'
```
Comment (dirkf) [inline youtube_dl/utils.py:2378]: Revert and fix `utils.bug_reports_message()` (2a):
```suggestion
        update_cmd = 'see  https://github.com/ytdl-org/youtube-dl/#user-content-installation  on how to update'
```

**Comment by salty-horse** [comment]
The video that caused the issues no longer does (probably because it was a live video on youtube, and after a while, I think they re-encode it in a different format.) But I forced the error, and it works fine.

My only suggestion is removing the whitespace before `please report this issue`, because it the error has a double space:
`if any subtitle tracks are missing,  please report this issue`

File: youtube_dl/utils.py [hunk 4]
```diff
@@ -2371,15 +2371,24 @@ def set_alpn_protocols(ctx):
...
     else:
-        update_cmd = 'see  https://yt-dl.org/update  on how to update'
-    msg = '; please report this issue on https://yt-dl.org/bug .'
-    msg += ' Make sure you are using the latest version; %s.' % update_cmd
-    msg += ' Be sure to call youtube-dl with the --verbose flag and include its complete output.'
-    return msg
+        update_cmd = 'see  https://github.com/ytdl-org/youtube-dl/#user-content-installation  on how to update'
+
+    msg = (
+        ' please report this issue on https://github.com/ytdl-org/youtube-dl/issues ,'
```
Comment (dirkf) [inline youtube_dl/utils.py:2381]: Remove initial space that is added in l.2391 if needed.
```suggestion
        'please report this issue on https://github.com/ytdl-org/youtube-dl/issues ,'
```

**Comment by dirkf** [comment]
Yes, the crash would only have been seen in this specific combination of circumstances:
* the selected format is a DASH manifest
* the manifest has subtitles
* the extractor calls the original yt-dl DASH extraction API `_extract_mpd_formats()`, rather than the newer yt-dlp API `_extract_mpd_formats_and_subtitles()` (this might be all yt-dl extractors at the moment).

Many thanks for your help!

**Pull request merged**